src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3985,37 +3985,6 @@ private MethodHandle getDirectMethodCommon(byte refKind, Class<?> refc, MemberNa
                 checkSecurityManager(refc, method);
             assert(!method.isMethodHandleInvoke());
 
-            if (refKind == REF_invokeSpecial &&
-                refc != lookupClass() &&
-                !refc.isInterface() &&
-                refc != lookupClass().getSuperclass() &&
-                refc.isAssignableFrom(lookupClass())) {
-                assert(!method.getName().equals("<init>"));  // not this code path
-
-                // Per JVMS 6.5, desc. of invokespecial instruction:
-                // If the method is in a superclass of the LC,
-                // and if our original search was above LC.super,
-                // repeat the search (symbolic lookup) from LC.super
-                // and continue with the direct superclass of that class,
-                // and so forth, until a match is found or no further superclasses exist.
-                // FIXME: MemberName.resolve should handle this instead.
-                Class<?> refcAsSuper = lookupClass();
-                MemberName m2;
-                do {
-                    refcAsSuper = refcAsSuper.getSuperclass();
-                    m2 = new MemberName(refcAsSuper,
-                                        method.getName(),
-                                        method.getMethodType(),
-                                        REF_invokeSpecial);
-                    m2 = IMPL_NAMES.resolveOrNull(refKind, m2, lookupClassOrNull(), allowedModes);
-                } while (m2 == null &&         // no method is found yet
-                         refc != refcAsSuper); // search up to refc
-                if (m2 == null)  throw new InternalError(method.toString());
-                method = m2;
-                refc = refcAsSuper;
-                // redo basic checks
-                checkMethod(refKind, refc, method);
-            }
```

Comment:
This code exists to locate the correct declaring class for an invokespecial. Since we don't need to find the declaring class, it's unnecessary.

I verified that a `lookupSpecial` method reference still behaves appropriately: if you do `findSpecial` on an ancestor class's method, where that method has been overridden by a parent class, the behavior of `invoke` continues to be the parent method, not the ancestor method.